### PR TITLE
Fixes test_client to use gen_test.

### PR DESCRIPTION
**Pull request by oglu**
The test checks whether the hook is called from the client. This should help with #872

### CODE REVIEW

File: circus/tests/test_client.py [hunk 1]
```diff
@@ -78,41 +77,51 @@ def test_client(self):
         yield self.stop_arbiter()
 
 
+TMP_FILE = './HOOK_CALLED'
+
+
 def long_hook(*args, **kw):
-    time.sleep(5)
+    with open(TMP_FILE, 'a'):
```
Comment (k4nar) [inline circus/tests/test_client.py]: Could you use a file created via `tempfile` instead? I don't really like creating files in the current directory for the tests, and it's not very side-effect proof.

Comment (oglu) [inline circus/tests/test_client.py]: I tried to do it without file but the way the hook is resolved and the lack of a way to pass arguments to the hook excludes any communication between the hook and the test. Even if we had a tempfile we wouldn't be able to communicate the file name/fd to the hook. 

I know it's a gross hack but I was unable to come up with something better without introducing things to the core.

Comment (k4nar) [inline circus/tests/test_client.py]: I think I found a cleaner way, by creating a temporary file, and deleting it in the hook:

```patch
diff --git i/circus/tests/test_client.py w/circus/tests/test_client.py
index 0380cb3..9a56a11 100644
--- i/circus/tests/test_client.py
+++ w/circus/tests/test_client.py
@@ -1,4 +1,5 @@
 import os
+import tempfile
 
 from tornado.testing import gen_test
 from tornado.gen import coroutine, Return
@@ -77,12 +78,11 @@ class TestClient(TestCircus):
         yield self.stop_arbiter()
 
 
-TMP_FILE = './HOOK_CALLED'
+_, tmp_filename = tempfile.mkstemp(prefix='test_hook')
 
 
 def long_hook(*args, **kw):
-    with open(TMP_FILE, 'a'):
-        pass
+    os.unlink(tmp_filename)
 
 
 class TestWithHook(TestCircus):
@@ -97,13 +97,12 @@ class TestWithHook(TestCircus):
 
     @gen_test
     def test_message_id(self):
-        if os.path.exists(TMP_FILE):
-            os.unlink(TMP_FILE)
-
         hooks = {'before_stop': ('circus.tests.test_client.long_hook', False)}
         testfile, arbiter = self.run_with_hooks(hooks)
         yield arbiter.start()
         try:
+            self.assertTrue(os.path.exists(tmp_filename))
+
             msg = make_message("numwatchers")
             resp = yield self.cli.call(msg)
             self.assertEqual(resp.get("numwatchers"), 1)
@@ -118,9 +117,10 @@ class TestWithHook(TestCircus):
             resp = yield self.cli.call(make_message("numwatchers"))
             self.assertEqual(resp.get("numwatchers"), 1)
 
-            self.assertTrue(os.path.exists(TMP_FILE))
+            self.assertFalse(os.path.exists(tmp_filename))
         finally:
-            os.unlink(TMP_FILE)
+            if os.path.exists(tmp_filename):
+                os.unlink(tmp_filename)
             arbiter.stop()
```

It's not perfect but at least is doesn't create any file in the current directory :) .

Comment (oglu) [inline circus/tests/test_client.py]: Shall I change the PR or you can do it ?

Comment (k4nar) [inline circus/tests/test_client.py]: You can change the PR (I can't commit on your branch).

Comment (oglu) [inline circus/tests/test_client.py]: Oh... When I applied the patch I remembered why I didn't use tempfile or a global var (for the same effect). The tempfile the hook sees, is different from the tempfile the test, as the module of the hook is resolved and loaded in a way that does not use the module of the test. Does this test pass from your side ?

Comment (k4nar) [inline circus/tests/test_client.py]: I had the same reflexion when writing it, but yes it does pass.

Comment (oglu) [inline circus/tests/test_client.py]: Strange. For me it fails at the last assertion (assertFalse for the existence of file)

Comment (oglu) [inline circus/tests/test_client.py]: Oh, I think I know. Try running only test_message_id

Comment (k4nar) [inline circus/tests/test_client.py]: It also works with `nosetests -vsx circus/tests/test_client.py:TestWithHook.test_message_id`.

Comment (oglu) [inline circus/tests/test_client.py]: Very strange. Indeed, it passes from cmdline (unittest/nosetest) but somehow fails at pycharm. Anyway, I've amended the PR.

Comment (k4nar) [inline circus/tests/test_client.py]: Well, it passed on Travis, so we'll say we're fine for now.

Comment (oglu) [inline circus/tests/test_client.py]: As we'd like to have a new release pushed, is there something else that's blocking a 0.12 ?

Comment (k4nar) [inline circus/tests/test_client.py]: I would like to take a look a #829 but it's not necessarily a blocker.

**Pull request merged**